doc/designs/passkeys.md

```diff
@@ -0,0 +1,246 @@
+by many companies or government agencies. Alternate and more secure
+solutions exist, among which the use of passkeys, where the private
+key is stored on the device and the server only needs to know the public
+key.
+
+The goal of this feature is to use a passkey to authenticate a user
+against IPA.
+
+The project will be jointly developed by SSSD and IPA:
+
+- IPA provides the interface to store the user's public credentials
+- IPA provides the interface to configure passkey settings
+- SSSD performs the actual authentication
+
+## Use Cases
+
+- The administrator or the user registers a passkey into IPA, associated
+to a user account. The registration requires the insertion of the key, or
+the string output provided by SSSD registration mechanism.
+- The user is then able to authenticate to any IPA enrolled host using the
+passkey. Authentication methods include console login, desktop login,
+or su access.
+
+## How to Use
+
+### Configuration of the passkey settings by the administrator
+
+The administrator is able to specify common settings that will apply:
+
+- require user verification during authentication (On/Off/Default):
+  - On: require user verification during authentication (PIN for instance).
+  - Off: do not require user verification during authentication.
+  - Default: fallback to the passkey’s default behavior.
+
+### Registration of credentials
+
+The user can register credentials for himself, or the admin (or any user with
+the permission "System: Manage User passkeys") can register
+credentials for another user.
+
+During the registration process, it is possible to specify
+- a COSE type: `es256`, `rs256` or `eddsa`
+- request user verification: true or false
+the authentication will force to execute the user verification check even if
+the passkey settings do not set this flag. If credentials are registered without
+the flag, the global passkey settings apply.
+
+### Authentication
+
+#### Console or desktop authentication
+
+The user has a passkey in his possession that was already registered to IPA
+and has physical access to a machine enrolled in IPA.
+
+At Gnome login, he types his username and inserts the device.
+
+At console login, he types his username and inserts the device.
+
+SSSD validates the credentials and checks that the passkey allows
+authentication.
+
+#### Su access
+
+The user passkeyuser has a passkey in his possession that was already
+registered to IPA and has physical access to a machine enrolled in IPA. He
+is already logged into the machine as a different user and wants to perform
+su to authenticate as passkeyuser.
+
+Inside a terminal, he inserts his device and enters the `su - passkeyuser`
+command.
+
+SSSD validates the credentials and checks that the passkey allows
+authentication.
+
+
+## Design
+
+### Configuration of the passkey settings
+
+A new LDAP entry stores the passkey configuration and needs a new objectclass
+and a new attributetype:
+```
+dn: cn=passkeyconfig,$BASEDN
```

Comment:
We store other configuration settings under `cn=etc,$BASEDN`:
- OTP defaults stored in `cn=otp,cn=etc,$BASEDN`
- IPA configuration is in `cn=ipaconfig,cn=etc,$BASEDN`
- topology plugin uses `cn=ipa,cn=etc,$BASEDN` tree

Perhaps passkey defaults should be in `cn=passkey,cn=etc,$BASEDN`?